docs/plugins-environment-variables.md

```diff
@@ -51,8 +46,300 @@ The plugins included with the development platform are:
+      resolvers: <array of resolvers>,
+      schemas: <array of schemas>,
+}
+```
+
+### i18n
+
+Use the `i18n` object to register translations.
+
+### mutations and queries
+
+The `mutations` and `queries` objects are maps of functions that are extended onto `context.mutations` and `context.queries`. These may be functions that are called from GraphQL resolvers of a similar name, or functions that are intended to be called only by other plugin code. Functions that modify data should be registered as `mutations` and all other functions should be registered as `queries`.
+
+### simpleSchemas
+
+The SimpleSchema package is used by many plugins to validate data, often before inserting or updating MongoDB documents. To allow other plugins to extend these schemas, some plugins register them such that they are accessible in a `preStartup` function.
+
+Here's an example of registering two schemas:
+
+```js
+simpleSchemas: {
+  Cart, CartItem;
+}
+```
+
+And a different plugin can then extend them in a `preStartup` function:
+
+```js
+export default function preStartup(context) {
+  context.simpleSchemas.CartItem.extend({
+    additionalCartItemField: String,
+  });
+}
+```
+
+(This could be done in a `startup` function, but because startup code sometimes validates against these schemas, it's safer to do it in a `preStartup` function.)
+
+## Developing API Plugins
+
+### Switching Reaction to Development Mode
+
+Now let’s say you want to make some changes to a service and be able to see those changes reflected in the running service. By default, any changes you make do not affect the running service because all services run in standard mode. It’s done this way because it is much faster to start a service in standard mode. To develop a service, you need to switch it to development mode.
+Development mode differs from standard mode in the following ways:
+
+- A generic Node.js development Docker image is used for the container rather than the published service image.
+- Your locally checked out project files, other than those under `node_modules`, are mirrored into the container by way of a Docker volume mount.
+- NPM packages are installed when the container starts and therefore reflect the project’s `package.json` file. (In other words, you can add additional NPM dependencies and then stop and restart the container.)
+- In the container, the project runs within nodemon, which will restart the app whenever you change any file in the project.
+
+Switching a project to development mode is easy.
+
+1. Start the whole system in standard mode using `make` command in the Reaction Development Platform directory.
+2. Run `make dev-<project folder name>` for one or more services to restart them in development mode. For example, `make dev-reaction` if you want to make API changes.
+
+But what exactly is this doing? Hopefully you don’t need to concern yourself with the details, but for those who want to know or who run into troubles, here’s the breakdown:
```

Comment:
Do we actually need this? I think no.

Reply:
@zenweasel I think the details of how the switch happens are not useful to folks who are developing plugins. These details are only useful internally to understand linking and therefore should be removed from this doc.

Reply:
Unclear on this comment since this is your PR?